### ISSUE TITLE
flaky test:  `drop_cache_rather_than_verify`

### ISSUE THREAD

**Issue by jugmac00** [zopefoundation/ZEO#179]
see https://github.com/zopefoundation/ZEO/pull/178/checks?check_run_id=2425920984

This error does not happen all the time = flaky test.

```
Failure in test /home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/tests/drop_cache_rather_than_verify.txt
Failed doctest test for drop_cache_rather_than_verify.txt
  File "/home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/tests/drop_cache_rather_than_verify.txt", line 0

----------------------------------------------------------------------
File "/home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/tests/drop_cache_rather_than_verify.txt", line 146, in drop_cache_rather_than_verify.txt
Failed example:
    for e in events:
        print(e)
Expected:
    (2, '', 'StaleCache')
Got nothing

...../opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/subprocess.py:689: ResourceWarning: unclosed file <_io.BufferedReader name=4>
  _cleanup()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/nagios.py:140: ResourceWarning: unclosed <socket.socket fd=9, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('0.0.0.0', 51580)>
  addr, options.output_metrics, options.status_path, options.time_units)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
......../home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/asyncio/client.py:880: ResourceWarning: unclosed resource <TCPTransport closed=False reading=True 0x7f1560003d90>
  loop.close()
/home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name='./1-1.zec'>
  gc.collect()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/runner/work/ZEO/ZEO/.tox/py37-uvloop/lib/python3.7/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.TextIOWrapper name='./1-1.zec.lock' mode='a+' encoding='UTF-8'>
  gc.collect()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
```

**Comment by d-maurer** [comment]
This behavior is really strange. By default, `zope-testrunner` reports all failed examples from a doctest. Unless `tox` runs `zope-testrunner` with some hidden options, the `drop_cache_rather_than_verify.txt` doctest contains a single failing example. Especially the example following the failed example succeeds: it checks the critical log message. This means, the `StaleCache` event was not observed but the `StaleCache` log message was observed. However, if one of them is issued, both of them are -- and this before "connected" is signaled and the cache is cleared. I do not yet see how this is possible.

**Comment by dataflake** [comment]
Fixed by #187
